src/presentation/http/router/upload.ts

```diff
@@ -0,0 +1,108 @@
+import type { FileTypes } from '@domain/entities/file';
+import type { Note, NotePublicId } from '@domain/entities/note.js';
+import type FileUploaderService from '@domain/service/fileUploader.service.js';
+import type { MultipartFile, MultipartValue } from '@fastify/multipart';
+import type { FastifyPluginCallback } from 'fastify';
+import type NoteService from '@domain/service/note.js';
+import { notEmpty } from '@infrastructure/utils/empty.js';
+
+/**
+ * Interface for upload router options
+ */
+interface UploadRouterOptions {
+  /**
+   * File uploader service
+   */
+  fileUploaderService: FileUploaderService;
+
+  /**
+   * Note service instance
+   */
+  noteService: NoteService;
+}
+
+const UploadRouter: FastifyPluginCallback<UploadRouterOptions> = (fastify, opts, done) => {
+  const { fileUploaderService } = opts;
+  const { noteService } = opts;
+
+  fastify.post<{
+    Body: {
+      /**
+       * File to upload
+       */
+      file: MultipartFile;
+
+      /**
+       * File type
+       */
+      type: MultipartValue<FileTypes>;
+
+      /**
+       * Note public id, if file is related to a note
+       */
+      noteId?: MultipartValue<NotePublicId>;
```

Comment:
seems hardcoded for NoteAttachement, use `location` instead

Reply:
done